lib/stat_tracker.rb

```diff
@@ -90,4 +90,65 @@ def average_goals_by_season
       average_of_goals_in_a_season(season.first.season)
     end
   end
+
+  def count_of_teams
+    @teams.length
+  end
+
+  def average_goals_by_team(team_id, hoa = nil)
+    goals = 0
+    games = 0
+    @game_teams.each do |game_team|
+      if hoa != nil
+        if game_team.team_id == team_id && game_team.hoa == hoa
+          goals += game_team.goals
+          games += 1
+        end
+      elsif game_team.team_id == team_id
+          goals += game_team.goals
+          games += 1
+      end
+    end
+    return 0 if games == 0
+    (goals.to_f / games.to_f).round(2)
+  end
+
+  def unique_team_ids
+    @game_teams.uniq { |game_team| game_team.team_id }.map{ |game_team| game_team.team_id }
```

Comment:
Either put uniq at the end of the map, or maybe do a group_by team_id and then call the key values?